modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/SchemaMigrator.java

```diff
@@ -1,130 +1,87 @@
+* Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.wso2.carbon.is.migration.service;
 
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
-import org.wso2.carbon.is.migration.client.internal.ISMigrationServiceDataHolder;
-import org.wso2.carbon.is.migration.util.Constants;
-import org.wso2.carbon.is.migration.util.ResourceUtil;
+import org.wso2.carbon.identity.core.migrate.MigrationClientException;
+import org.wso2.carbon.is.migration.internal.ISMigrationServiceDataHolder;
+import org.wso2.carbon.is.migration.util.Constant;
+import org.wso2.carbon.is.migration.util.Utility;
 import org.wso2.carbon.utils.dbcreator.DatabaseCreator;
 
-import javax.sql.DataSource;
 import java.io.BufferedReader;
-import java.io.File;
 import java.io.FileInputStream;
 import java.io.InputStream;
 import java.io.InputStreamReader;
 import java.sql.Connection;
-import java.sql.DatabaseMetaData;
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.sql.SQLWarning;
 import java.sql.Statement;
 import java.util.StringTokenizer;
 
-public class MigrationDatabaseCreator {
+/**
+ * Migrator implementation for Schema migration.
+ */
+public class SchemaMigrator extends Migrator {
+
+    private static final Log log = LogFactory.getLog(SchemaMigrator.class);
+
+    public static final String LOCATION = "location" ;
+
+    private String location ;
 
-    private static Log log = LogFactory.getLog(MigrationDatabaseCreator.class);
-    private DataSource dataSource;
-    private DataSource umDataSource;
     private Connection conn = null;
     private Statement statement;
     private String delimiter = ";";
 
-    public MigrationDatabaseCreator(DataSource dataSource, DataSource umDataSource) {
-
-//        super(dataSource);
-        this.dataSource = dataSource;
-        this.umDataSource = umDataSource;
-    }
 
-    /**
-     * Execute Migration Script
-     *
-     * @throws Exception
-     */
-    public void executeIdentityMigrationScript() throws Exception {
+    @Override
+    public void migrate() throws MigrationClientException {
+        this.location = getMigratorConfig().getParameterValue(LOCATION);
 
+        log.info(Constant.MIGRATION_LOG + "Executing Identity Migration Scripts.");
         try {
-            conn = dataSource.getConnection();
+            conn = getDataSource().getConnection();
```

Comment:
can we use try-with-resources?

Reply:
conn is defined in class level and using it in other places. So cant use try with resources since that need to instantiate variable inside try